main.php

```diff
@@ -18,18 +18,19 @@ function init() {
 		add_action( 'posts_clauses', array( __CLASS__, 'posts_clauses' ), 10, 2 );
 	}
 
-	function manage_cookie() {
+	public static function manage_cookie() {
 		// Default conditions
 		$clear_condition = false;
 		$read_condition = is_singular();
 		$set_condition = !is_404();
+		$var = new self();
 
 		if ( apply_filters( 'smarter_nav_clear', $clear_condition ) )
-			self::clear_cookie();
+			$var->clear_cookie();
 		elseif ( apply_filters( 'smarter_nav_read', $read_condition ) )
-			self::read_cookie();
+			$var->read_cookie();
 		elseif ( apply_filters( 'smarter_nav_set', $set_condition ) )
-			self::set_cookie();
+			$var->set_cookie();
```

Comment:
Instead of using the `$var = new self();` hack, it would be cleaner to just make all the methods static.